File: fizzbuzz/scala/fizzbuzz.scala
object FizzBuzz extends Application {
  val fizzbuzz = {
    def loop(i: Int): Stream[String] =
      ((i % 3, i%5) match {
        case (0,0) => "FizzBuzz"
        case (0,_) => "Fizz"
        case (_,0) => "Buzz"
        case _ => i.toString()
      }) #:: loop(i + 1)
    loop(1)
  }

  fizzbuzz take 100 foreach println
}

File: fizzbuzz/rust/fizzbuzz.rs
fn main() {
  for i in range(1, 101) {
    println!("{}",
      match (i % 3, i % 5) {
        (0, 0) => "FizzBuzz".to_owned(),
        (0, _) => "Fizz".to_owned(),
        (_, 0) => "Buzz".to_owned(),
        _ => i.to_str()
      }
    )
  }
}
